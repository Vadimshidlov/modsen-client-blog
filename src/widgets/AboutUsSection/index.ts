export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
