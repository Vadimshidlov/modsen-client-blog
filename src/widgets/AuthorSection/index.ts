export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
