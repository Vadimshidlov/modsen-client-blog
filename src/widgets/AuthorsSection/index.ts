export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
