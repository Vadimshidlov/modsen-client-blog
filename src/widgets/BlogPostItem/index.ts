export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
