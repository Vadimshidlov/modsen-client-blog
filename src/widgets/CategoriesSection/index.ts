export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
