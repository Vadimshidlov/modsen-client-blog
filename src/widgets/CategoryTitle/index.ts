export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
