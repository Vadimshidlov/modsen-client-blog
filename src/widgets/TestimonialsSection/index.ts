export { TestimonialsSection } from '@/widgets/TestimonialsSection/ui/TestimonialsSection';
