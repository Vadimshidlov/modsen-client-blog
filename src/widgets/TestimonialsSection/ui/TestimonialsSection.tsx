'use client';

import React from 'react';
import { Inter } from 'next/font/google';
import { useTranslations } from 'next-intl';
import { FeedBackSwiper } from '@/widgets/TestimonialsSection/ui/FeedBackSwiper';
import container from '@/shared/styles/container.module.scss';
import styles from '@/widgets/TestimonialsSection/ui/TestimonialsSection.module.scss';

const inter = Inter({
  subsets: ['latin'],
  weight: '600',
  variable: '--font-inter',
});

export function TestimonialsSection() {
  const t = useTranslations('TestimonialsSection');

  return (
    <section className={`${styles.testimotalsSection} ${container.container}`}>
      <div className={styles.contentContainer}>
        <article className={styles.infoBlock}>
          <h3 className={`${inter.variable} ${styles.infoSubTitle}`}>{t('subTitle')}</h3>
          <h2 className={styles.infoTitle}>{t('title')}</h2>
          <span className={styles.infoText}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.
          </span>
        </article>
        <div className={styles.contentSeparator} />
        <FeedBackSwiper />
      </div>
    </section>
  );
}
