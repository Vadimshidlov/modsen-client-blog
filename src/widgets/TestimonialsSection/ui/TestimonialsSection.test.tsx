import { render } from '@testing-library/react';
import { TestimonialsSection } from '@/widgets/TestimonialsSection/';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('TestimonialsSection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<TestimonialsSection />);
    expect(container).toMatchSnapshot();
  });
});
