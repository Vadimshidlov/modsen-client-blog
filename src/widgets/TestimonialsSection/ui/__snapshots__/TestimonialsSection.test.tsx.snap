// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestimonialsSection tests should matches snapshot 1`] = `
<div>
  <section
    class="testimotalsSection container"
  >
    <div
      class="contentContainer"
    >
      <article
        class="infoBlock"
      >
        <h3
          class="variable infoSubTitle"
        >
          Mocked subTitle
        </h3>
        <h2
          class="infoTitle"
        >
          Mocked title
        </h2>
        <span
          class="infoText"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.
        </span>
      </article>
      <div
        class="contentSeparator"
      />
      <div
        class="swiperWrapper"
        data-testid="swiper-wrapper"
      >
        <article
          class="messagesBlock"
          data-testid="swiper-block"
          style="transform: translateX(0%);"
        >
          <div
            class="swiperItem"
          >
            <h3
              class="messagesText"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua.
            </h3>
            <div
              class="authorInfoBlock"
            >
              <img
                alt="author-avatar"
                class="authorInfoAvatar"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <div
                class="authorInfoText"
              >
                <a
                  href="/authors/5"
                >
                  <span
                    class="authorInfoName"
                  >
                    Jonathan Vallem
                  </span>
                </a>
                <span
                  class="authorInfoAddress"
                >
                  New york, USA
                </span>
              </div>
            </div>
          </div>
          <div
            class="swiperItem"
          >
            <h3
              class="messagesText"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua.
            </h3>
            <div
              class="authorInfoBlock"
            >
              <img
                alt="author-avatar"
                class="authorInfoAvatar"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <div
                class="authorInfoText"
              >
                <a
                  href="/authors/5"
                >
                  <span
                    class="authorInfoName"
                  >
                    Dianne Russell
                  </span>
                </a>
                <span
                  class="authorInfoAddress"
                >
                  USA
                </span>
              </div>
            </div>
          </div>
          <div
            class="swiperItem"
          >
            <h3
              class="messagesText"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua.
            </h3>
            <div
              class="authorInfoBlock"
            >
              <img
                alt="author-avatar"
                class="authorInfoAvatar"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <div
                class="authorInfoText"
              >
                <a
                  href="/authors/5"
                >
                  <span
                    class="authorInfoName"
                  >
                    Leslie Alexander
                  </span>
                </a>
                <span
                  class="authorInfoAddress"
                >
                  TEXAS
                </span>
              </div>
            </div>
          </div>
        </article>
      </div>
      <div
        class="sliderController"
      >
        <button
          class="sliderButton"
          data-testid="swiper-left-button"
        >
          <img
            alt="swiper-left-btn"
            class="authorInfoAvatar"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
        <button
          class="sliderButton"
        >
          <img
            alt="swiper-right-btn"
            class="authorInfoAvatar"
            data-nimg="1"
            data-testid="swiper-right-button"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </button>
      </div>
    </div>
  </section>
</div>
`;
